pkgs/applications/version-management/gitlab/remove-hardcoded-locations.patch

```diff
@@ -96,14 +96,31 @@ index e7a8d08..834ecaf 100644
      # When you run tests we clone and setup gitlab-shell
 diff --git a/lib/gitlab/app_logger.rb b/lib/gitlab/app_logger.rb
 index 8e4717b..abfe2e4 100644
---- a/lib/gitlab/app_logger.rb
-+++ b/lib/gitlab/app_logger.rb
-@@ -1,7 +1,7 @@
- module Gitlab
-   class AppLogger < Gitlab::Logger
-     def self.file_name
--      'application.log'
-+      ENV["GITLAB_APPLICATION_LOG_PATH"]
+--- a/lib/gitlab/logger.rb
++++ b/lib/gitlab/logger.rb
+@@ -8,21 +8,22 @@
+       build.info(message)
      end
  
-     def format_message(severity, timestamp, progname, msg)
+     def self.read_latest
+-      path = Rails.root.join("log", file_name)
++      path = File.join(ENV['GITLAB_APPLICATION_LOG_PATH'], file_name)
```

Comment:
Huh, this looks weird, your change makes GITLAB_APPLICATION_LOG_PATH a path to .../application.log (a file) but then joins it with "file_name"? Doing the "join" sounds more appropriate when GITLAB_APPLICATION_LOG_PATH is a directory path and not a file.